llvm/lib/Target/RISCV/MCTargetDesc/RISCVBaseInfo.h

```diff
@@ -624,6 +624,15 @@ inline unsigned encodeRlist(MCRegister EndReg, bool IsRV32E = false) {
   }
 }
 
+inline static unsigned encodeRlistNumRegs(unsigned NumRegs) {
+  assert(NumRegs > 0 && NumRegs < 14 && NumRegs != 12 &&
+         "Unexpected number of registers");
+  if (NumRegs == 13)
+    --NumRegs;
```

Comment:
```suggestion
    return RLISTENCODE::RA_S0_S11;
```

I think this would be clearer.